### DISCUSSION TITLE
Renaming

### DISCUSSION THREAD

**Discussion by vertigo220** [doublecmd/doublecmd#1978] [Suggestions]
Just spent a lot of time in DC the other night renaming a bunch of files and found the process problematic for two reasons:

1. Slow-double-clicking very frequently didn't work, and almost every time I'd have to click 3, 4, or even 5 times to get it to finally activate the renaming. And once or twice it ended up opening the file because I clicked too fast. That's an issue I sometimes have regardless of the file manager I'm using, but it showed that the first issue isn't that I was clicking too slowly. It was just not responding properly to my clicks.
2. Clicking outside the file name when renaming doesn't save the new name like it does in other file managers. It only saves if hitting enter. This is not only much slower, but I didn't realize it until I had "renamed" several files, forcing me to go back and check and do many of them again since they reverted to the original name without me realizing when I'd clicked out of them. There should at least be an option to switch this behavior.

**Comment by Skif-off** [comment]
Please, if you have problem or question, do not forget to specify the version of DC ("Help" in the main menu > About > copy).

1 ) I use this feature a lot and can't confirm. I've had problems, but it's always been a mouse problem (I love my mouse, so I healed it with a soldering iron).

2 ) [RenameConfirmMouse](https://doublecmd.github.io/doc/en/configxml.html#RenameConfirmMouse).

> like it does in other file managers

For example? (Except Windows Explorer.)

> It only saves if hitting enter.

This is not true: `Enter` or the rename confirmation button (to the right of the edit field).

**Reply by vertigo220** [reply]
v1.1.20~607

1) Definitely not the mouse, as this is the only program that has trouble. I've used other file managers extensively without running into this issue, as well as numerous other programs.

2) Perfect. I tried searching the options and didn't see anything, forgot there are more settings aside from the options window. That, combined with the "[difficulty](#1980)" of editing the config xml is why I feel all options should be in the program itself, even if the xml contents are just displayed in their own tab which can be edited and the changes applied which would automatically restart the program.

> This is not true: `Enter` or the rename confirmation button (to the right of the edit field).

Ok, true, but I was saying of the two options, which are the two common methods, as in, between hitting enter and clicking outside, only enter works. Sorry that wasn't clear.

**Reply by Skif-off** [reply]
2 ) `RenameConfirmMouse` is mentioned in the [corresponding subsection](https://doublecmd.github.io/doc/en/help.html#rename_files).
(Most of these parameters are explicitly mentioned in the description of the corresponding functions/actions.)
